roles/openstack/tasks/cleanup.yaml

```diff
@@ -0,0 +1,73 @@
+---
+# (c) Copyright IBM Corporation
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+- name: Fetch the path of neerali openstack systems provisioned file
+  ansible.builtin.set_fact:
+    neerali_openstack_systems_provisoned_path: >-
+      {{
+        (neerali_openstack_artifactdir, 'neerali_openstack_provisioned_systems.yaml') |
+        ansible.builtin.path_join
+      }}
+
+- name: Check if the neerali openstack systems provisoned file exists
+  ansible.builtin.stat:
+    path: "{{ neerali_openstack_systems_provisoned_path }}"
+  register: neerali_systems_provisioned_stat
+
+- name: Load the yaml content if the file exists
+  ansible.builtin.include_vars:
+    file: "{{ neerali_openstack_systems_provisoned_path }}"
+  when: neerali_systems_provisioned_stat.stat.exists
+
+- name: Delete openstack networks
+  vars:
+    neerali_network_action: delete
+    neerali_network_payload: "{{ neerali_systems_layout.networks }}"
+  ansible.builtin.include_tasks: network.yaml
+  when: neerali_systems_layout.networks is defined
+
+- name: Delete servers
+  vars:
+    neerali_server_action: delete
+  ansible.builtin.include_tasks: server.yaml
+  when: neerali_openstack_systems_provisioned is defined
+
+- name: Delete volumes
+  vars:
+    neerali_volume_action: delete
+  ansible.builtin.include_tasks: volume.yaml
+  loop: "{{ neerali_openstack_systems_provisioned }}"
+  loop_control:
+    loop_var: compute_node
+  when: neerali_openstack_systems_provisioned is defined
+
+- name: Delete a neerali openstack systems provisioned file
+  ansible.builtin.file:
+    path: "{{ neerali_openstack_systems_provisoned_path }}"
+    state: absent
+  when: neerali_systems_provisioned_stat.stat.exists
+
+- name: Ensure workspace is cleaned
+  ansible.builtin.file:
+    path: "{{ item }}"
+    state: absent
+  loop:
+    - "{{ neerali_openstack_basedir }}"
+    - "{{ neerali_openstack_artifactdir }}"
+    - "{{ neerali_openstack_logdir }}"
```

Comment:
`neerali_openstack_basedir` points to `~/neerali-data`... if removed, then we might break other dependencies....